src/pkg/services/m365/api/drive_pager.go

```diff
@@ -0,0 +1,317 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/alcionai/clues"
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/graph/api"
+	"github.com/alcionai/corso/src/pkg/logger"
+	abstractions "github.com/microsoft/kiota-abstractions-go"
+	"github.com/microsoftgraph/msgraph-sdk-go/drives"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites"
+	"github.com/microsoftgraph/msgraph-sdk-go/users"
+)
+
+// ---------------------------------------------------------------------------
+// item pager
+// ---------------------------------------------------------------------------
+
+type driveItemPager struct {
+	gs      graph.Servicer
+	driveID string
+	builder *drives.ItemRootDeltaRequestBuilder
+	options *drives.ItemRootDeltaRequestBuilderGetRequestConfiguration
+}
+
+func NewItemPager(
+	gs graph.Servicer,
+	driveID, link string,
+	fields []string,
+) *driveItemPager {
+	pageCount := maxPageSize
+
+	headers := abstractions.NewRequestHeaders()
+	preferHeaderItems := []string{
+		"deltashowremovedasdeleted",
+		"deltatraversepermissiongaps",
+		"deltashowsharingchanges",
+		"hierarchicalsharing",
+	}
+	headers.Add("Prefer", strings.Join(preferHeaderItems, ","))
```

Comment:
we should make a note/PR to combine this with the code in Exchange that makes the `Prefer` header kv-pair

Reply:
will combine in a follow-up